### ISSUE TITLE
Code annotations with extra comments have a wrong ID

### ISSUE THREAD

**Issue by JulioV** [squidfunk/mkdocs-material#2444]
- [x ] I've read the [contribution guidelines][1] and agree with them

__I've found a bug and checked that ...__

- [x ] ... the problem doesn't occur with the default MkDocs template
- [ x] ... the problem is not in any of my customizations (CSS, JS, template)
- [ x] ... the documentation does not mention anything about my problem
- [ x] ... there are no open or closed issues that are related to my problem

## Description

Comments that are not annotations are counted as one, so the next annotation ID is wrong.

For example, for:
```yaml
PHONE_ACCELEROMETER: # (1)
ANOTHER: 1 # not an annotation
CONTAINER: accelerometer # (2)
```

### Expected behavior

```html
PHONE_ACCELEROMETER: (1)
ANOTHER: 1 # not an annotation
CONTAINER: accelerometer (2)
```

### Actual behavior

```html
PHONE_ACCELEROMETER: (1)
ANOTHER: 1 # not an annotation
CONTAINER: accelerometer (3)
```

### Steps to reproduce the bug

1. ...
2. ...
3. ...

### Package versions

* Python: `python --version` 3.7.9
* MkDocs: `mkdocs --version` 1.1.2
* Material: `pip show mkdocs-material | grep -E ^Version` Version: 7.0.6+insiders.2.3.1

### Project configuration

``` yaml
The contents of your mkdocs.yml
```

### System information

* OS: Mac
* Browser: Firefox

  [1]: https://github.com/squidfunk/mkdocs-material/blob/master/CONTRIBUTING.md

**Comment by squidfunk** [comment]
Thanks for reporting! Confirmed!

**Comment by squidfunk** [comment]
... and fixed in the latest commit on the Insiders repository:

<img width="713" alt="Bildschirmfoto 2021-03-14 um 19 45 14" src="https://user-images.githubusercontent.com/932156/111080297-18595780-84fe-11eb-8a34-de1c4c712fe4.png">

**Comment by JulioV** [comment]
Awesome, thanks!

**Comment by squidfunk** [comment]
The bugfixes were just released as part of Insiders 2.4.0, which also adds support for [custom admonition icons](http://localhost:8000/reference/admonitions/#changing-the-icons)
